src/input.cpp

```diff
@@ -62,6 +117,29 @@ int Input::KeyNewPressed(unsigned int buttons) const
 	return KeyPressed(buttons) && (oldpad.buttons & buttons) != buttons;
 }
 
+int Input::KeyNewPressedNoRepeat(unsigned int buttons) const {
```

Comment:
I suggest making this the default behavior of `Input::KeyNewPressed`. You could then either have an optional parameter `repeat = false`, or have a new function `Input::KeyNewPressedRepeat` although it's a bit unclear what `Repeat` means.